### PR TITLE
feat: add logic to obtain the user's courses dynamically

### PR DESCRIPTION
**Pull request by BryanttV**
### Description
This PR adds a feature that allows a user's courses to be dynamically fetched from the platform. Until now, course fetching was static. Also, some docstring improvements were also included.

### CODE REVIEW

File: sample-skill/lambda/alexa/utils.py [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+        if the request is successful, None otherwise.
+    """
+    response = requests.post(
+        url, data=request_data, headers=headers, timeout=MAX_TIMEOUT
+    )
+
+    if response.status_code == HTTPStatus.OK:
+        return response.json()
+
+    return None
```
Comment (mariajgrimaldi) [inline sample-skill/lambda/alexa/utils.py]: You could implement a make_request method with signature: URL, data, headers, method

Comment (mariajgrimaldi) [inline sample-skill/lambda/alexa/utils.py]: you can return an empty dict and use .get() so it always returns none when there's an empty response. That could be useful for the methods below. So you don't have to `if X else None`

Comment (BryanttV) [reply sample-skill/lambda/alexa/utils.py]: Done

Comment (BryanttV) [reply sample-skill/lambda/alexa/utils.py]: Done

**Review by mariajgrimaldi — APPROVED** [review]

**Pull request merged**